scripts/fetch_litecore.py

```diff
@@ -0,0 +1,309 @@
+        The path of the folder to download the LiteCore artifacts into
+    """
+
+    variant_pair = variant_to_pair(variant)
+    subdirectory = subdirectory_for_variant(variant_pair[0], variant_pair[1]) if has_platform else f"{variant_pair[0]}/{variant_pair[1]}"
+    return Path(output_base).joinpath(subdirectory).resolve()
+
+def download_variant(download_folder: str, variant: str, debug: bool, output_base: str) -> int:
+    """Performs the download and extraction of LiteCore artifacts
+    
+    Parameters
+    ----------
+     download_folder : str
+        The URL of the folder containing the variant to be downloaded
+    variant : str
+        The name of the variant (e.g. windows-win64)
+    debug: bool
+        If true, download the debug variant instead of the release one
+    output_base : str
+        The path to the base output directory for the downloaded files
+
+    Returns
+    -------
+    int
+        0 on success, 1 on failure (to be used for running tally of failures)
+    """
+
+    filename = filename_for_platform(variant, debug)
+    download_url = f"{download_folder}/{filename}"
+    download_path = calculate_download_path(variant, output_base)
+    print(f"--- Downloading {filename} to {download_path}...")
+    
+    os.makedirs(download_path, exist_ok=True)
+
+    full_path = f"{download_path}/{filename}"
+    try:
+        urllib.request.urlretrieve(download_url, full_path)
+    except HTTPError as e:
+        print(f"!!! Failed: {e.code}")
+        return 0
+
+    print(f"--- Extracting {filename}...")
+    if filename.endswith("tar.gz"):
+        with tarfile.open(full_path, "r:gz") as tar:
+            tar.extractall(download_path)
+    else:
+        with zipfile.ZipFile(full_path) as zip:
+            zip.extractall(download_path)
+    
+    os.remove(full_path)
+    return 1
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description='Fetch a specific prebuilt LiteCore')
+    parser.add_argument('--variants', nargs='+', type=str, help='A space separated list of variants to download', required=True)
+    parser.add_argument('--debug', action='store_true', help='If specified, download debug variants')
+    parser.add_argument('--dry-run', action='store_true', help='Check for existience of indicated artifacts, but do not perform download')
+    parser.add_argument('--sha', type=str, help="The SHA to download.  If not provided, calculated based on the provided CE and (optionally) EE repos.  Required if CE not specified.")
+    parser.add_argument('--ce', type=str, help="The path to the CE LiteCore repo.  Required if SHA not specified.")
+    parser.add_argument('--ee', type=str, help="The path to the EE LiteCore repo")
+    parser.add_argument('--path', type=str, help="The path in which the platform specific extensions to this script are defined (platform_fetch.py).  If a relative path is passed, it will be relative to fetch_litecore.py.  By default it is the current working directory.",
```

Comment:
Would love either a -q --quiet or a -v verbose flag that allowed me to silence all but essential output.

Reply:
What would you consider essential?  I guess the output of a dry run should not be silenced but other than that there is not really a need for any output.